File: Task_Manager_C++/TaskManager.cpp
#include "TaskManager.h"
#include <algorithm>

TaskManager::TaskManager() : tasks()
{
}

void TaskManager::addTask(string title,string desc){
    if(title.empty()){
        tasks.push_back(Task());
    }else if(desc.empty()){
        tasks.push_back(Task(title));
    }else{
        tasks.push_back(Task(title,desc));
    }
}

Task TaskManager::getTask(int id) {
    int searchId = 0;
    
    while (searchId != id) {
        if (tasks[searchId].getId() == id) {
            return tasks[searchId];
        } else if (searchId >= tasks.size()) {
            cout << "Id de la Task non attribué" << endl;
            break;
        } else {
            searchId++;
        }
    }
    
    // Aucune tâche correspondante n'a été trouvée, retourne une tâche vide par exemple
    return Task(); 
}

void TaskManager::removeTask(int id) {
    for (int i=0;i<tasks.size();i++) {
        if (tasks[i].getId() == id ) {
            tasks.erase(tasks.begin()+i+1);
        }
    }
}

void TaskManager::removeAllTasks(){
    tasks.clear();
    
}

bool TaskManager::OkTask(int id) {
    for (const Task& task : tasks) {
        if (task.getId() == id && task.isTaskDone()) {
            return true;
        }
    }
    return false;
}

void TaskManager::afficheTasks(){
    for (const Task& task : tasks) {
        cout << "<---><---><---><---><---><---><---><--->" << endl;
        cout << "ID : " <<task.getId() << endl;
        cout << task.getTitle() << " : \n" << endl;
        cout << "Etat : "; 
        if(OkTask(task.getId())){cout << "Fini\n"<<endl;}else{cout << "En cours\n"<<endl;} 
        cout << task.getDescription() << "\n" << endl;
        cout << "<---><---><---><---><---><---><---><--->\n" << endl;
    }
}

File: Task_Manager_Java/TaskManager.java
import java.util.ArrayList;
import java.util.List;

/**
 * La classe TaskManager gère une liste de tâches.
 */
public class TaskManager {
    private List<Task> tasks;

    /**
     * Constructeur de la classe TaskManager.
     * Initialise la liste de tâches.
     */
    public TaskManager() {
        this.tasks = new ArrayList<>();
    }

    /**
     * Ajoute une nouvelle tâche à la liste de tâches.
     * Si le nom est vide, une tâche par défaut est ajoutée.
     * @param nom Le nom de la tâche.
     * @param desc La description de la tâche.
     * @param etat L'état de la tâche (terminée ou non).
     */
    public void addTask(String nom, String desc, boolean etat) {
        if (!nom.equals("")) {
            tasks.add(new Task(nom, desc, etat));
        } else {
            tasks.add(new Task());
        }
    }

    /**
     * Supprime une tâche de la liste de tâches.
     * Réduit également le nombre total de tâches.
     * @param index L'indice de la tâche à supprimer.
     */
    public void removeTask(int index) {
        tasks.remove(index);
        Task.setNbTask(Task.getNbTask() - 1);
    }

    /**
     * Marque une tâche comme terminée.
     * @param index L'indice de la tâche à marquer comme terminée.
     */
    public void OkTask(int index) {
        tasks.get(index).setEtat(true);
    }

    /**
     * Affiche les détails de toutes les tâches dans la liste.
     */
    public void AfficheTask() {
        for (int i = 0; i < tasks.size(); i++) {
            System.out.println("<---><---><---><---><---><---><---><--->");
            System.out.println(tasks.get(i).getNom() + " :\n");
            System.out.println("Etat : " + (tasks.get(i).getEtat() ? "Fini" : "En cours") + "\n");
            System.out.println(tasks.get(i).getDescription() + "\n");
            System.out.println("<---><---><---><---><---><---><---><--->\n");
        }
    }
}
